### PR TITLE
Data Migration Service (DMS) sdk for public preview.

### PR DESCRIPTION
**Pull request by hitenjava**
## Description
- Added public preview sdk for Data Migration Service (DMS) public preview release.
- The sdk is based on swagger spec here - https://github.com/Azure/azure-rest-api-specs-pr/tree/datamigration-v2017-04-15/specification/datamigration/resource-manager/Microsoft.DataMigration/2017-11-15-preview

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [x] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [x] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [x] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code.
- [x] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

### CODE REVIEW

File: src/SDKs/DataMigration/DataMigration.Tests/DataMigration.Tests.csproj [hunk 1]
```diff
@@ -0,0 +1,58 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
```
Comment (shahabhijeet) [inline src/SDKs/DataMigration/DataMigration.Tests/DataMigration.Tests.csproj]: @hitenjava model your project looking at other projects in the repo (e.g. https://github.com/Azure/azure-sdk-for-net/blob/psSdkJson6/src/SDKs/BatchAI/BatchAI.Tests/BatchAI.Tests.csproj

File: src/SDKs/DataMigration/DataMigration.Tests/Helpers/Utilities.cs [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+            {
+                handler.IsPassThrough = true;
+                dmClient = context.GetServiceClient<DataMigrationServiceClient>(handlers: handler);
+            }
+            return dmClient;
+        }
+
+        private static HttpClientHandler GetHandler()
+        {
+#if DNX451
```
Comment (shahabhijeet) [inline src/SDKs/DataMigration/DataMigration.Tests/Helpers/Utilities.cs]: @hitenjava we do not support this target? These projects are no longer DNX based projects.
We only support one compile constant and that is FullNetFx

File: src/SDKs/DataMigration/Management.DataMigration/Microsoft.Azure.Management.DataMigration.csproj [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    <AssemblyName>Microsoft.Azure.Management.DataMigration</AssemblyName>
+    <PackageTags>Microsoft Azure resource management;Data Migration Service;REST HTTP client;azureofficial;windowsazureofficial;netcore451511</PackageTags>
+  </PropertyGroup>
+  <PropertyGroup>
+    <TargetFrameworks>net452;netstandard1.4</TargetFrameworks>
+  </PropertyGroup>
+  
+  <!--
+
+  <ItemGroup>
```
Comment (shahabhijeet) [inline src/SDKs/DataMigration/Management.DataMigration/Microsoft.Azure.Management.DataMigration.csproj]: @hitenjava please remove commented code

File: src/SDKs/DataMigration/Management.DataMigration/Microsoft.Azure.Management.DataMigration.csproj [hunk 4]
```diff
@@ -0,0 +1,39 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+  <Import Project="$([MSBuild]::GetPathOfFileAbove('AzSdk.reference.props'))" />
+  <PropertyGroup>
+    <PackageId>Microsoft.Azure.Management.DataMigration</PackageId>
+    <Description>Provides developers with libraries for the Data Migration Service (DMS) under Azure Resource manager to deploy an instance of DMS. Create, Delete, Create project, Start Migration, and more.</Description>
+    <Version>0.1.0-preview</Version>
+    <AssemblyName>Microsoft.Azure.Management.DataMigration</AssemblyName>
+    <PackageTags>Microsoft Azure resource management;Data Migration Service;REST HTTP client;azureofficial;windowsazureofficial;netcore451511</PackageTags>
```
Comment (shahabhijeet) [inline src/SDKs/DataMigration/Management.DataMigration/Microsoft.Azure.Management.DataMigration.csproj]: @hitenjava add <packageReleaseNotes>
e.g. https://github.com/Azure/azure-sdk-for-net/blob/psSdkJson6/src/SDKs/BatchAI/Management.BatchAI/Microsoft.Azure.Management.BatchAI.csproj

File: src/SDKs/DataMigration/DataMigration.Tests/ScenarioTests/CRUDProjectTests.cs [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+using Microsoft.Azure.Management.Resources;
+using Microsoft.Rest.ClientRuntime.Azure.TestFramework;
+using System.Net;
+using Xunit;
+
+namespace DataMigration.Tests.ScenarioTests
+{
+    public class CRUDProjectTests : CRUDDMSTestsBase
+    {
+        [Fact]
```
Comment (shahabhijeet) [inline src/SDKs/DataMigration/DataMigration.Tests/ScenarioTests/CRUDProjectTests.cs]: @hitenjava use the latest resource manager 1.6.0 at least to record all your tests.

**Review by shahabhijeet — CHANGES_REQUESTED** [review]
Tests are failing in CI

File: src/SDKs/DataMigration/DataMigration.Tests/ScenarioTests/CRUDProjectTests.cs (hunk 5 above)
Comment (hitenjava) [reply src/SDKs/DataMigration/DataMigration.Tests/ScenarioTests/CRUDProjectTests.cs]: @shahabhijeet - You mean Microsoft.Azure.ResourceManager? I only see v1.1.0-preview as the latest.

File: src/SDKs/DataMigration/DataMigration.Tests/DataMigration.Tests.csproj (hunk 1 above)
Comment (hitenjava) [reply src/SDKs/DataMigration/DataMigration.Tests/DataMigration.Tests.csproj]: Done.

File: src/SDKs/DataMigration/DataMigration.Tests/Helpers/Utilities.cs (hunk 2 above)
Comment (hitenjava) [reply src/SDKs/DataMigration/DataMigration.Tests/Helpers/Utilities.cs]: Done.

File: src/SDKs/DataMigration/Management.DataMigration/Microsoft.Azure.Management.DataMigration.csproj (hunk 4 above)
Comment (hitenjava) [reply src/SDKs/DataMigration/Management.DataMigration/Microsoft.Azure.Management.DataMigration.csproj]: Done.

File: src/SDKs/DataMigration/Management.DataMigration/Microsoft.Azure.Management.DataMigration.csproj (hunk 3 above)
Comment (hitenjava) [reply src/SDKs/DataMigration/Management.DataMigration/Microsoft.Azure.Management.DataMigration.csproj]: Done.

File: src/SDKs/DataMigration/DataMigration.Tests/ScenarioTests/CRUDProjectTests.cs (hunk 5 above)
Comment (rohit-joy) [reply src/SDKs/DataMigration/DataMigration.Tests/ScenarioTests/CRUDProjectTests.cs]: @shahabhijeet Where can we find the resource manager 1.6.0 to reference?

Comment (hitenjava) [reply src/SDKs/DataMigration/DataMigration.Tests/ScenarioTests/CRUDProjectTests.cs]: Had a call with Abhijeet, link to resource manager - https://www.nuget.org/packages/Microsoft.Azure.Management.ResourceManager/1.6.0-preview

I will update tests to use this.

**Review by shahabhijeet — APPROVED** [review]
Looks good, please reopen a new PR once your spec is approved and merged into public spec repo.

**Comment by shahabhijeet** [comment]
@hitenjava closing this PR beucase you will need a separate PR that has public repo spec referenced.

**Pull request closed without merging**